### PR TITLE
[ts] Initial commit of TypeScript definitions

### PR DESCRIPTION
**Pull request by buggy**
This PR is a first attempt at the TypeScript definitions. It has been scraped together from the API documentation and some experimentation with the actual API. 

It includes:

* A definition for the Shopify class as well as definitions for public and private app options passed to it.
* All of the API call are defined. Where types haven't been defined `any` is used.
* Most of the objects returned by the Shopify API are defined
* Some of the create/update API calls also have definitions for their parameters

@lpinca If you're happy to include this in the main project then I'm willing to maintain it.

@stephengardner I'd appreciate a second set of eyes over the definition if you have time.

Moving forward the plan is to expand the definitions to additional API's as required or as time permits.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,1987 @@
...
+    payment: {
+        count: (checkoutToken: any) => Promise<number>;
+        create: (checkoutToken: any, params: any) => Promise<any>;
+        get: (checkoutToken: any, id: number) => Promise<any>;
+        list: (checkoutToken?: any) => Promise<any>;
+    };
+    policy: {
+        list: (params?: any) => Promise<Shopify.IPolicy[]>;
+    };
+    priceRules: {
```
Comment (lpinca) [inline index.d.ts]: If I'm not wrong this should be `priceRule`.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,1987 @@
...
+        "themes/delete" |
+        "themes/publish" |
+        "themes/update";
+
+    type WebhookFormat = "json" | "xml";
+
+    export interface IWebhook {
+        address: string;
+        created_at: string;
+        fields: s
```
Comment (lpinca) [inline index.d.ts]: Please add a new line.

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,1987 @@
...
+    interface IImage {
+        created_at: string;
+        height: number;
+        src: string;
+        updated_at?: string;
+        width: number;
+    }
+
+    interface IBase64Image {
+        attrachment: string;
```
Comment (lpinca) [inline index.d.ts]: `attachment` ?

**Comment by lpinca** [comment]
Can you please include this in the [`files`](https://github.com/MONEI/Shopify-api-node/blob/2d372bd2381d8ad59df6fbd602c43b353f285a91/package.json#L12-L16) array?

**Comment by lpinca** [comment]
> @lpinca If you're happy to include this in the main project then I'm willing to maintain it.

Yes that would be great. I wonder if @alexandresaiz is ok with adding you as collaborator to the project.

File: index.d.ts (hunk 2 above)
Comment (lpinca) [reply index.d.ts]: Not sure why but this comment seems misplaced, I meant new line at the end of file.

**Comment by alexandresaiz** [comment]
@lpinca fyi, @buggy has been added with write perm.

**Comment by lpinca** [comment]
Awesome, thank you.

**Comment by lpinca** [comment]
I'll keep this open for a couple of days in case anyone want to take a look.

**Comment by lpinca** [comment]
Released version 2.10.1, thank you @buggy.

**Pull request merged**